File: Tutorial4-4ComponentBuildersMultipleModules/src/main/java/com/example/tutorial4_4componentbuildersmultiplemodules/MainActivity.java
package com.example.tutorial4_4componentbuildersmultiplemodules;

import android.content.SharedPreferences;
import android.hardware.SensorManager;
import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.example.tutorial4_4componentbuildersmultiplemodules.model.SensorController;
import com.example.tutorial4_4componentbuildersmultiplemodules.model.ToastMaker;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity implements SensorController.OnSensorEventChangeListener {

    @Inject
    SharedPreferences sharedPreferences;

    @Inject
    ToastMaker toastMaker;

    @Inject
    SensorController sensorController;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ((MyApplication) getApplication()).getAppComponent().inject(this);
/*
        ((MyApplication) getApplication()).getAppComponentWithBind().inject(this);
*/
        toastMaker.showToast("sharedPreferences " + sharedPreferences);
        sensorController.setSensorEventChangeListener(this);

    }

    @Override
    protected void onResume() {
        super.onResume();
        sensorController.onResume(SensorManager.SENSOR_DELAY_UI);
    }

    @Override
    protected void onPause() {
        super.onPause();
        sensorController.onPause();
    }

    @Override
    public void getOrientationValues(float azimuth, float pitch, float roll) {
        TextView textView = findViewById(R.id.result);
        textView.setText("Azimuth: " + azimuth + ", float: " + pitch + ", roll: " + roll);
    }
}

File: Tutorial8-1DaggerHilt-Basics/src/main/java/com/smarttoolfactory/tutorial8_1daggerhilt_basics/MainActivity.kt
package com.smarttoolfactory.tutorial8_1daggerhilt_basics

import android.content.SharedPreferences
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.smarttoolfactory.tutorial8_1daggerhilt_basics.model.SimpleDependency
import com.smarttoolfactory.tutorial8_1daggerhilt_basics.model.ToastMaker
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var simpleDependency: SimpleDependency

    @Inject
    lateinit var sharedPreferences: SharedPreferences

    @Inject
    lateinit var toastMaker: ToastMaker

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val tvInfo = findViewById<TextView>(R.id.tvInfo)

        findViewById<TextView>(R.id.tvInfo).text =
                "ApplicationModule sharedPreferences: ${sharedPreferences.hashCode()}\n" +
                        "ApplicationModule ToastMaker: ${toastMaker.hashCode()}\n" +
                        "Constructor dependency simpleDependency: ${simpleDependency.hashCode()}"

        toastMaker.showToast("Injected from AppModule")
    }
}